paddle_billing/Entities/Event.py

```diff
@@ -7,14 +7,15 @@
 from paddle_billing.Entities.Events import EventTypeName
 
 from paddle_billing.Notifications.Entities.Entity import Entity as NotificationEntity
+from paddle_billing.Notifications.Entities.UndefinedEntity import UndefinedEntity
 
 
 @dataclass
 class Event(Entity, ABC):
     event_id: str
     event_type: EventTypeName
     occurred_at: datetime
-    data: NotificationEntity | dict
+    data: NotificationEntity | UndefinedEntity
```

Comment:
Just wondering if `EntityDict` has any place here? Does it still provide value if it's not defined as part of any return typing?

Reply:
`EntityDict` just ensures that `NotificationEntity` and `UndefinedEntity` have a common interface, I think it's helpful to declare that `Entity` or `UndefinedEntity` could be returned here. We could instead change this to `data: EntityDict` and if we need to do `isinstance(event.data, EntityDict)` we can annotate with `@runtime_checkable`?